griptape/utils/chat.py

```diff
@@ -16,22 +17,39 @@ class Chat:
     prompt_prefix: str = field(default="Q: ", kw_only=True)
     response_prefix: str = field(default="A: ", kw_only=True)
     output_fn: Callable[[str], None] = field(
-        default=Factory(lambda: print), kw_only=True
+        default=lambda x: print(x, end=""), kw_only=True
     )
+    stream: bool = field(default=False, kw_only=True)
+
+    def start(self, initial_turn: str = None) -> None:
+        if initial_turn:
+            self.output_fn(initial_turn + "\n")
+
+        self.structure.stream = self.stream
+        self.structure.prompt_driver.stream = self.stream
+        if self.stream:
+            self.structure = Stream(self.structure)
 
-    def start(self) -> None:
         if self.intro_text:
-            self.output_fn(self.intro_text)
+            self.output_fn(self.intro_text + "\n")
         while True:
             question = input(self.prompt_prefix)
 
             if question.lower() in self.exit_keywords:
-                self.output_fn(self.exiting_text)
-
+                self.output_fn(self.exiting_text + "\n")
                 break
             else:
-                self.output_fn(self.processing_text)
+                self.output_fn(self.processing_text + "\n")
 
-            self.output_fn(
-                f"{self.response_prefix}{self.structure.run(question).output.to_text()}"
-            )
+            if self.stream:
+                self.output_fn(f"{self.response_prefix}")
+                self.structure.structure.event_listeners = (
+                    []
+                )  # To address https://github.com/griptape-ai/griptape/issues/408
+                for chunk in self.structure.run(question):
+                    self.output_fn(chunk.value)
```

Comment:
I think the `output_fn` implementation should be different for `stream=True` vs `stream=False`. For `stream=False` we can keep the old implementation. For `stream=True` we can use the new implementation.

Reply:
As discussed above, in some places we need \n and when streaming we need to suppress it. I think any modifications to the output_fn on the fly end up presuming the output_fn is 'print'. Would it be better to let users have the flexibility to specify 2 output_fn so we don't have to assume what they always be printing to console?